test/integrations/destinations/amazon_audience/router/data.ts

```diff
@@ -0,0 +1,226 @@
+            },
+            {
+              destination,
+              message: {
+                type: 'record',
+                action: 'insert',
+                fields: { email: 'email3@abc.com' },
+                channel: 'sources',
+                context: {},
+                recordId: '3',
+              },
+              metadata: generateMetadata(3),
+            },
+            {
+              destination,
+              message: {
+                type: 'record',
+                action: 'delete',
+                fields: { email: 'email4@abc.com' },
+                channel: 'sources',
+                context: {},
+                recordId: '4',
+              },
+              metadata: generateMetadata(4),
+            },
+            {
+              destination,
+              message: {
+                type: 'record',
+                action: 'delete',
+                fields: {
+                  email: 'email5@abc.com',
+                },
+                channel: 'sources',
+                context: {},
+                recordId: '5',
+              },
+              metadata: generateMetadata(5),
+            },
+            {
+              destination,
+              message: {
+                type: 'identify',
+                context: {},
+                recordId: '1',
+              },
+              metadata: generateMetadata(6),
+            },
+          ],
+          destType: 'amazon_audience',
+        },
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: {
+          output: [
+            {
+              batched: true,
+              batchedRequest: {
+                version: '1',
+                type: 'REST',
+                method: 'POST',
+                endpoint: '',
+                headers: {
+                  'Amazon-Advertising-API-ClientId': 'dummyClientId',
+                  'Content-Type': 'application/json',
+                  Authorization: 'Bearer dummyAccessToken',
+                },
+                params: {},
+                body: {
+                  JSON: {
+                    associateUsers: {
+                      patches: [
+                        {
+                          op: 'add',
+                          path: '/EXTERNAL_USER_ID-Rudderstack_8a6ae15122001229edb8866f56e342af12ae8187203c3e3b33931743e7c0c48d/audiences',
+                          value: ['dummyId'],
+                        },
+                      ],
+                    },
+                    createUsers: {
+                      records: [
+                        {
+                          externalId:
+                            'Rudderstack_8a6ae15122001229edb8866f56e342af12ae8187203c3e3b33931743e7c0c48d',
+                          hashedRecords: [
+                            {
+                              email: 'email1@abc.com',
+                            },
+                            {
```

Comment:
For Update shouldnt we delete first then insert?

Reply:
Nothing is visible from dashboard so I guess we should be good inserting it directly as it seems to be a upsert endpoint

Reply:
ok